common/StringVector.hpp

```diff
@@ -111,64 +128,84 @@ class StringVector
 
     /// Tokenize single-char delimited values until we hit new-line or the end.
     static StringVector tokenize(const char* data, const std::size_t size,
-                                 const char delimiter = ' ')
+                                 const char delimiter = ' ', bool ignoreQuotes = true)
     {
         if (size == 0 || data == nullptr || *data == '\0')
             return StringVector();
 
         std::vector<StringToken> tokens;
-        tokenize(data, size, delimiter, tokens);
+        tokenize(data, size, delimiter, tokens, ignoreQuotes);
         return StringVector(std::string(data, size), std::move(tokens));
     }
 
     /// Tokenize single-char delimited values until we hit new-line or the end.
-    static StringVector tokenize(const std::string& s, const char delimiter = ' ')
+    static StringVector tokenize(const std::string& s, const char delimiter = ' ', bool ignoreQuotes = true)
     {
         if (s.empty())
             return StringVector();
 
         std::vector<StringToken> tokens;
-        tokenize(s.data(), s.size(), delimiter, tokens);
+        tokenize(s.data(), s.size(), delimiter, tokens, ignoreQuotes);
         return StringVector(s, std::move(tokens));
     }
 
     /// Tokenize by the delimiter string.
-    static StringVector tokenize(const std::string& s, const char* delimiter, int len = -1)
+    /// Ignoring delimiters that are within quotation marks, unless ignoreQuotes is set to true
+    static StringVector tokenize(const std::string& s, const char* delimiter, int len = -1, bool ignoreQuotes = true)
     {
         if (s.empty() || len == 0 || delimiter == nullptr || *delimiter == '\0')
             return StringVector();
 
         if (len < 0)
             len = std::strlen(delimiter);
 
-        std::size_t start = 0;
-        std::size_t end = s.find(delimiter, start);
+        std::string delimiterString(delimiter);
+        if(delimiterString.find('\'') != std::string::npos || delimiterString.find('"') != std::string::npos)
```

Comment:
Please use clang-format (you can configure your editor to run it on *modified* lines upon save).